### PR TITLE
Allow ordered expectations

### LINKED ISSUE #108: Allow ordered expectations

### PR DESCRIPTION
**Pull request by jeremystucki**
Resolves #108

### CODE REVIEW

File: src/internal/method.rs [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+}
+
 #[derive(Debug)]
 pub struct Method<'mock, A, R>
 where
     A: for<'args> ArgumentsMatcher<'args>,
 {
     name: &'static str,
-    calls: Vec<MethodCall<'mock, A, R>>,
+    calls: VecDeque<MethodCall<'mock, A, R>>,
```
Comment (bash) [inline src/internal/method.rs]: Can you go back to a `Vec`?

File: src/internal/method.rs [hunk 2]
```diff
@@ -54,6 +68,37 @@ where
...
+    ) -> Result<R, CallError<'mock, 'a, A, R>> {
+        match self.call_order {
+            ExpectedCallOrder::Sequentially => {
+                self.handle_call_with_sequentially_ordered_expectations(arguments)
+            }
+            ExpectedCallOrder::Unordered => self.handle_call_with_unordered_expectations(arguments),
+        }
+    }
+
+    fn handle_call_with_sequentially_ordered_expectations<'a>(
```
Comment (bash) [inline src/internal/method.rs]: Can you test the new behaviour in this file also?

Comment (bash) [reply src/internal/method.rs]: Add at least a test that verifies that a method gives an error when no more calls are expected.

File: tests/sequential_calls.rs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+trait Foo {
+    fn bar(&self) -> bool;
+}
+
+#[test]
+fn sequential_calls() {
+    let mut mock = FooMock::new();
+
+    mock.expect_bar().returns(true);
+    mock.expect_bar().returns(false);
```
Comment (bash) [inline tests/sequential_calls.rs]: Also test, that other method's expectations are still unordered.

File: src/internal/method.rs (hunk 1 above)
Comment (jeremystucki) [reply src/internal/method.rs]: c77fbba

File: tests/sequential_calls.rs (hunk 3 above)
Comment (jeremystucki) [reply tests/sequential_calls.rs]: 87070c1

File: src/internal/method.rs (hunk 2 above)
Comment (jeremystucki) [reply src/internal/method.rs]: 87070c1

**Pull request merged**